### PR TITLE
Feature/4 save state and load state

### PR DESCRIPTION
**Pull request by kevintumbo**
created tests for save state and load state

### CODE REVIEW

File: Tests/database_tests.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.create_room("living", ["Kampala"])
+        self.dojo.add_person('Ian', 'pyr' ,'FELLOW', 'Y')
+        self.dojo.add_person('Joel', 'Ortiz' ,'FELLOW', 'Y')
+        self.dojo.add_person('Armin', 'Buren' ,'STAFF')
+        msg = self.dojo.save_state('test')
+        self.assertTrue(os.path.isfile("test.db"))
+        self.assertEqual(msg, "You have saved data to the database")
```
Comment (jimmykimani) [inline Tests/database_tests.py]: I was advised checking if the file has been created isn't the best way to test this functionality. You don't know if it has the content you want. You could import sqlite3 (if it's an sqlite database) and assert that the columns contain the content you expect.

File: Tests/database_tests.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        os.remove("january_data.db")
+
+    def test_save_state_succesfully_stores_data_in_database(self):
+
+        """ Test successful storage of data from app into database using the save_state method """
+
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.create_room("living", ["Kampala"])
+        self.dojo.add_person('Ian', 'pyr', 'FELLOW', 'Y')
```
Comment (mnjaggah) [inline Tests/database_tests.py]: You could also use the load_people from txt file to add people.

File: Tests/database_tests.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        self.dojo.add_person('Armin', 'Buren', 'STAFF')
+        msg = self.dojo.save_state('test')
+        self.assertTrue(os.path.isfile("test.db"))
+        self.assertEqual(msg, "You have saved data to the database")
+        conn = sqlite3.connect("test.db")
+        c = conn.cursor()
+        name = ('Ian pyr',)
+        c.execute('SELECT * FROM people WHERE name=?', name)
+        row = c.fetchone()
+        assert row[1] == 'Ian pyr'
```
Comment (mnjaggah) [inline Tests/database_tests.py]: Good idea to assert data in the db columns

**Review by mnjaggah — COMMENTED** [review]
Tests look comprehensive

File: dojo.py [hunk 4]
```diff
@@ -305,7 +306,116 @@ def load_people(self, filename):
...
+            for space in self.total_rooms:
+                if individual in space.occupants and space.type == "office":
+                    person.office_allocated = space.room_name
+                if individual in space.occupants and space.type == "living":
+                    person.living_space_allocated = space.room_name
+
+            new_session.add(person)
+            new_session.commit()
+
+        return 'You have saved data to the database'
```
Comment (oscamaina) [inline dojo.py]: There is chance to duplicate database entries for one that has previously got some information, I would suggest you have a filter statement somewhere in your lines before adding new person or room where you make sure the new item added isn't similar to one that already exists

Comment (kevintumbo) [reply dojo.py]: a filter statement. thanks. let me implement that.

File: Tests/database_tests.py (hunk 2 above)
Comment (kevintumbo) [reply Tests/database_tests.py]: that would reduce number of lines. will do. thanks

**Pull request closed without merging**